egs/ami/s5/local/chain/run_tdnn_py.sh

```diff
@@ -0,0 +1,197 @@
+echo "$0 $@"  # Print the command line for logging
+
+. cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh
+
+if ! cuda-compiled; then
+  cat <<EOF && exit 1
+This script is intended to be used with GPUs but you have not compiled Kaldi with CUDA
+If you want to use GPUs (and have them), go to src/, and configure and make on a machine
+where "nvcc" is installed.
+EOF
+fi
+
+# The iVector-extraction and feature-dumping parts are the same as the standard
+# nnet3 setup, and you can skip them by setting "--stage 8" if you have already
+# run those things.
+
+suffix=
+if [ "$speed_perturb" == "true" ]; then
+  suffix=_sp
+fi
+dir=exp/chain/$mic/tdnn${affix:+_$affix} # Note: _sp will get added to this if $speed_perturb == true.
+dir=${dir}$suffix
+train_set=train$suffix
+treedir=exp/$mic/chain/train_tree$suffix
+lang=data/lang_chain
+
+# if we are using the speed-perturbed data we need to generate
+# alignments for it.
+local/nnet3/run_ivector_common.sh --stage $stage \
+                                  --mic $mic \
+                                  --use-ihm-ali $use_ihm_ali \
+                                  --use-sat-alignments $use_sat_alignments || exit 1;
+
+# we still support this option as all the TDNN, LSTM, BLSTM systems were built
+# using tri3a alignments
+if [ $use_sat_alignments == "true" ]; then
+  gmm=tri4a
+else
+  gmm=tri3a
+fi
+
+if [ $use_ihm_ali == "true" ]; then
+  gmm_dir=exp/ihm/$gmm
+  mic=${mic}_cleanali
+  ali_dir=${gmm_dir}_${mic}_train_parallel_sp_ali
+else
+  gmm_dir=exp/$mic/$gmm
+  ali_dir=${gmm_dir}_${mic}_train_sp_ali
+fi
+
+final_lm=`cat data/local/lm/final_lm`
+LM=$final_lm.pr1-7
+graph_dir=$gmm_dir/graph_${LM}
+
+
+
+if [ $stage -le 10 ]; then
+  # Get the alignments as lattices (gives the CTC training more freedom).
+  # use the same num-jobs as the alignments
+  nj=$(cat $ali_dir/num_jobs) || exit 1;
+  steps/align_fmllr_lats.sh --nj $nj --cmd "$train_cmd" data/$mic/$train_set \
+    data/lang $ali_dir exp/$mic/train_lats$suffix
+  rm exp/$mic/train_lats$suffix/fsts.*.gz # save space
+fi
+
+
+if [ $stage -le 11 ]; then
+  # Create a version of the lang/ directory that has one state per phone in the
+  # topo file. [note, it really has two states.. the first one is only repeated
+  # once, the second one has zero or more repeats.]
+  rm -rf $lang
+  cp -r data/lang $lang
+  silphonelist=$(cat $lang/phones/silence.csl) || exit 1;
+  nonsilphonelist=$(cat $lang/phones/nonsilence.csl) || exit 1;
+  # Use our special topology... note that later on may have to tune this
+  # topology.
+  steps/nnet3/chain/gen_topo.py $nonsilphonelist $silphonelist >$lang/topo
+fi
+
+if [ $stage -le 12 ]; then
+  # Build a tree using our new topology.
+  steps/nnet3/chain/build_tree.sh --frame-subsampling-factor 3 \
+      --leftmost-questions-truncate $leftmost_questions_truncate \
+      --cmd "$train_cmd" 9000 data/$mic/$train_set $lang $ali_dir $treedir
+fi
+
+if [ $stage -le 13 ]; then
+  echo "$0: creating neural net configs";
```

Comment:
I agree that 9k is too much.  I suggest to take whatever value was used in
the regular  nnet2 or nnet3 recipes and decrease it by about 10%.


On Fri, Mar 4, 2016 at 5:14 PM, Vijayaditya Peddinti <
notifications@github.com> wrote:

> In egs/ami/s5/local/chain/run_tdnn_py.sh
> <https://github.com/kaldi-asr/kaldi/pull/554#discussion_r55096889>:
>
> > +  silphonelist=$(cat $lang/phones/silence.csl) || exit 1;
> > +  nonsilphonelist=$(cat $lang/phones/nonsilence.csl) || exit 1;
> > +  # Use our special topology... note that later on may have to tune this
> > +  # topology.
> > +  steps/nnet3/chain/gen_topo.py $nonsilphonelist $silphonelist >$lang/topo
> > +fi
> > +
> > +if [ $stage -le 12 ]; then
> > +  # Build a tree using our new topology.
> > +  steps/nnet3/chain/build_tree.sh --frame-subsampling-factor 3 \
> > +      --leftmost-questions-truncate $leftmost_questions_truncate \
> > +      --cmd "$train_cmd" 9000 data/$mic/$train_set $lang $ali_dir $treedir
> > +fi
> > +
> > +if [ $stage -le 13 ]; then
> > +  echo "$0: creating neural net configs";
>
> Could you use the same TDNN config as the normal local/nnet3/run_tdnn.sh
> script.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/554/files#r55096889>.
>